src/main/java/Mel.java

```diff
@@ -0,0 +1,204 @@
+                parseAndAddItem(line, userList);
+            }
+            scanner.close();
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found: " + filePath);
+        }
+    }
+
+    private static void parseAndAddItem(String line, List userList) {
+        String[] parts = line.split(" \\| ");
+        String taskType = parts[0]; // T, D, E
+        boolean isDone = parts[1].equals("X");
+        String taskDescription = parts[2];
+
+
+        switch (taskType) {
+        case "T":
+            Todo todoTask = new Todo(taskDescription);
+            if (isDone) {
+                todoTask.markAsDone();
+            }
+            userList.itemArrayList.add(todoTask);
+            break;
+
+        case "D":
+            String deadlineDate = parts[3]; // Ignore remaining parts
+            Deadline deadlineTask = new Deadline(taskDescription, deadlineDate);
+            if (isDone) {
+                deadlineTask.markAsDone();
+            }
+            userList.itemArrayList.add(deadlineTask);
+            break;
+
+        case "E":
+            String eventStart = parts[3]; // Start date/time
+            String eventEnd = parts[4];   // End date/time
+            Event eventTask = new Event(taskDescription, eventStart, eventEnd);
+            if (isDone) {
+                eventTask.markAsDone();
+            }
+            userList.itemArrayList.add(eventTask);
+            break;
+
+        default:
+            System.out.println("Invalid task type in file: " + taskType);
+            break;
+        }
+    }
+
+
+    private static void saveListToFile(List userList) {
+        try {
+            writeToFile(LIST_FILE_PATH, userList.getFormattedTasks()); // getFormattedTasks returns a formatted String
+        } catch (IOException e) {
+            System.out.println("An error occurred while saving the list.");
+        }
+    }
+
+    private static void getUserInput(Scanner in, List userList) {
+        String line;
+        while (true) {
+            line = getLine(in);
+
+            if (isBye(line)) {
+                printByeMessage();
+                printHorizontalLine();
+                break;
+
+            } else if (isList(line)) {
+                userList.printList();
+                printHorizontalLine();
+
+            } else if (isMark(line)) {
+                userList.markItem(line);
+                printHorizontalLine();
+                saveListToFile(userList);
+
+            } else if (isUnmark(line)) {
+                userList.unmarkItem(line);
+                printHorizontalLine();
+                saveListToFile(userList);
+
+            } else if (isDelete(line)) {
+                userList.deleteItem(line);
+                printHorizontalLine();
+                saveListToFile(userList);
+
+            } else {
+                userList.addItem(line);
+                printHorizontalLine();
+                saveListToFile(userList);
+
```

Comment:
Consider using a switch case here instead of using many methods and if-else cases. This could be harder to maintain and does not look as clean